pkg/scheduler/cache/cache.go

```diff
@@ -462,6 +466,7 @@ func (cache *schedulerCache) RemoveNode(node *v1.Node) error {
 		delete(cache.nodes, node.Name)
 	}
 
+	cache.nodeTree.RemoveNode(node)
```

Comment:
I think we should catch error and then return it here.

Reply:
Returning an error here would be a change of behavior. A caller might assume that when an error is returned, the node is not removed from the cache, while the node is already removed in the above lines. Given this and the fact that nodeTree.RemoveNode returns an error only when the node is not found in the tree, I don't think we should return an error here.

Reply:
Ohh, you are right, then shouldn't we reverse the order, first we would remove the node from the tree and then remove it from cache or atleast log it instead of supressing it?

Reply:
Not sure about the reverse order. Given that NodeTree is a part of cache, I would prefer to keep it in sync with other cache structures. So, if a node is in the cached NodeInfo struct, I would prefer it to be kept in NodeTree as well.
Added logging to NodeTree.Remove.